site/content/articles/monitoring-multiple-openshift-clusters.adoc

```diff
@@ -0,0 +1,74 @@
+---
+title: "Monitoring Multiple Openshift Clusters"
+date: 2019-11-08T08:05:32+01:00
+authors: ["redhat-cop"]
+categories:
+  - openshift
+  - operations
+tags:
+  - research
+draft: true
+---
+= Monitoring Multiple Openshift Clusters
+Andrea Ferraresi <andrea.ferraresi@redhat.com>
+:toc: macro
+:toc-title:
+
+include::site/layouts/variables.adoc[]
+
+How to deal with monitoring of multiple Openshift Cluster: the challenges and the different approaches on scaling 
+
+toc::[]
+
+== Introduction to prometheus
+We all know Prometheus to be a very simple, very easy to use, piece of software:
+
+It is just a scraping server, this means that reads its metrics from a clear text page exposed by a web server on a specific port, metrics are generated from "exporters" which fills those pages. Once metrics are scraped are stored on disc for visualization. 
+
+It is not in the scope of Prometheus even to proper visualize those metrics, the UI is very simple and you would use Grafana for the graphing. 
+
+To keep things dead simple, there isn't any replication mechanism either.
+
+== The problems with prometheus
+
+This simplicity though comes with a cost: 
+
+*A*. Grafana can only query one Prometheus at a time which means that your monitoring server is a single point of failure: Having a single Prometheus instance per cluster is a single point of failure. 
+
+***
+Any configuration error or hardware failure could potentially result in the loss of important insights. Even a simple rollout could be a small disruption in the metric collection because a restart can be significantly longer than the scraping interval.
+***
+
+*B*. Storing on disc, on simple text files is not effective: reliability, availability, storage 
+space and speed can be an issue when you want to store a lot of data 
+
+== How do we fix them
+Let's try to fix the issues one at a time: 
+
+In order to fix *A* we could use one cool feature of the sofware that we call "Federation", this basically means that you could use an extra prometheus to query other isntances, this means that each of your cluster could run one prometheus, data would be queried by another instance installed outside of the cluster, in that way each cluster could be monitor on its own, with the extra possibility to have some sort of a supervisor. 
```

Comment:
This is a really long sentence and you could bring in some full stops instead of the commas. First could come after 'other instances' (note there is a typo on instances above)/ I'd suggest opening the next sentence with a different turn of phrase as 'this means' is used already and really close to it. Perhaps it could be dropped and just begin with 'Each of your clusters...'. Next full stop could be after 'outside of the cluster.'. 

Could be monitor on its own <-- could be '**a** monitor perhaps?